Optimisation: Simplify parsing M862

gudnimg opened the pull request:
This gets rid of float conversion.
We just need to check the 6th character when M862 is detected.

Change in memory:
Flash: -20 bytes
SRAM: 0 bytes

gudnimg committed:
Simplify parsing M862

This gets rid of float conversion.
We just need to check the 6th character when M862 is detected. This
character gives us the digit after the dot.

Change in memory:
Flash: -20 bytes
SRAM: 0 bytes

wavexx commented on Firmware/Marlin_main.cpp:
```diff
@@ -8034,10 +8034,9 @@ SERIAL_PROTOCOLPGM("\n\n");
-          nCommand=(uint8_t)(modff(code_value_float(),&nDummy)*10.0+0.5);
-          switch((ClPrintChecking)nCommand)
+    {
+        ClPrintChecking nCommand = static_cast<ClPrintChecking>(strtol(strchr_pointer+5, NULL, 10));
```
We need to check the string length here before we do strchr_pointer+5. We only know the string is at least 4 chars long including null.

gudnimg replied:
@wavexx here is my attempt to address this issue: [0bd85cc](https://github.com/prusa3d/Prusa-Firmware/pull/3795/commits/0bd85cc23ebb3b34ea2f985fc06957da546e903e) It looks like checking the string length or looking for a null byte consumes much more flash than we are trying to save. But with this commit we still save 20 bytes. 👍

gudnimg committed:
M862: determine decimal through math rather than checking for null byte/pointer.

This way doesn't add any more flash memory.
Calculating the string length or checking for a null byte
will end up increasing flash consumption overall.
